proms-resources/backend/src/main/java/io/uhndata/cards/proms/internal/importer/PatientLocalStorage.java

```diff
@@ -0,0 +1,332 @@
+    private static final String JCR_SQL = "JCR-SQL2";
+
+    /** Sling property name for the question reference field on an Answer node. */
+    private static final String QUESTION_FIELD = "question";
+
+    /** Sling property name for the value field on an Answer node. */
+    private static final String VALUE_FIELD = "value";
+
+    /** Provides access to resources. */
+    private final ResourceResolver resolver;
+
+    /** Details about a patient that we will parse and store in JCR. */
+    private JsonObject patientDetails;
+
+    PatientLocalStorage(JsonObject patientDetails, ResourceResolver resolver)
+    {
+        this.resolver = resolver;
+        this.patientDetails = patientDetails;
+    }
+
+    /**
+     * Store the patient details given to us.
+     */
+    public void store()
+    {
+        String mrn = this.patientDetails.getString("mrn");
+        Set<String> nodesToCheckin = new HashSet<String>();
+        try {
+            // Update information about the patient
+            Resource patient = getOrCreateSubject(mrn, "/SubjectTypes/Patient", null, this.resolver, nodesToCheckin);
+            Resource patientInfo = getOrCreateForm(patient, "/Questionnaires/Patient information/", this.resolver,
+                nodesToCheckin);
+            updatePatientInformationForm(patientInfo, this.patientDetails, this.resolver);
+
+            // Update information about the visit ("appointment" is used interchangably here)
+            JsonObject appointmentDetails = this.patientDetails.getJsonObject("nextAppointment");
+            Resource visit = getOrCreateSubject(appointmentDetails.getString("fhirID"),
+                "/SubjectTypes/Patient/Visit/", patient, this.resolver, nodesToCheckin);
+            Resource visitInfo = getOrCreateForm(visit, "/Questionnaires/Visit information/", this.resolver,
+                nodesToCheckin);
+            updateVisitInformationForm(visitInfo, appointmentDetails, this.resolver);
+
+            final Session session = this.resolver.adaptTo(Session.class);
+            session.save();
+
+            final VersionManager vm = session.getWorkspace().getVersionManager();
+            nodesToCheckin.forEach(node -> {
+                try {
+                    vm.checkin(node);
+                } catch (RepositoryException e) {
+                    LOGGER.warn("Failed to check in node {}: {}", node, e.getMessage(), e);
+                }
+            });
+        } catch (IOException e) {
+            LOGGER.error("Could not save patient {}: {}", mrn, e.getMessage(), e);
+        } catch (RepositoryException e) {
+            LOGGER.error("Could not save patient {}: {}", mrn, e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Grab a subject of the specified type, or create it if it doesn't exist.
+     * @param identifier Identifier to use for the subject
+     * @param parentTypePath path to a SubjectType node for this subject
```

Comment:
Sounds more like `subjectTypePath` than `parent`, right?

```suggestion
     * @param subjectTypePath path to a SubjectType node for this subject
```